docs/rfcs/0000-persistent-logging-component.md

```diff
@@ -0,0 +1,111 @@
+
+In the UI for a selected recording, the rows of data collected with options to filter by the various attributes and their combinations (a description of these types can be see in the Table Schema section):
+
+ - recording ID
+ - tag
+ - severity
+
+The logs can then be shared with the Android system share service to be attached to an email, Github issue, clipboard, etc. This possibly not ideal, see the Drawbacks section for more details.
+
+Similar to `lib-crash` most of the basic UI will be written into an abtract activity/fragment.
+
+## Reference-level explanation
+
+### Table Schema
+
+The schema below describes the primary log table that will be used to store logs:
+
+| Column       | Description                                                  | Type    |
+| ------------ | ------------------------------------------------------------ | ------- |
+| id           | Primary key for each log line.                               | Long    |
+| timestamp    | The time when the log event had occurred.                    | Long    |
+| recording_id | A random UUID generated at the start of a log recording.     | String  |
+| severity     | The priority of the log statement. See Android's [Log](https://developer.android.com/reference/android/util/Log#ERROR) constants. | Char    |
+| tag          | Used to identify the source of a log message.  It usually identifies the class or activity where the log call occurs. | String? |
+| log          | The message you would like logged.                           | String? |
+| throwable    | Optional. An exception to log.                               | String? |
+
+A secondary table is for storing metadata for each recording:
+
+| Column          | Description                            | Type |
+| --------------- | -------------------------------------- | ---- |
+| recording_id    | Primary key; foreign key for log table | Long |
+| start_timestamp | When the recording was started.        | Long |
+| end_timestamp   | When the recording was ended.          | Long |
+
+
+
+## Drawbacks
+
+- Sharing logs through an intent is fallible and can throw exceptions if the contents are more than the Android OS can handle.
+- User-initiated recording would mean we can lose logs that are not easily reproducible. We can always re-consider changing this in the future.
+
+## Future Work
+
+- We can request an API to stream GeckoView logs to our `LogSink` .
+- Additional useful device info can be prepended to the logs. For example:
+  - Delete a log entry
+  - System information
+  - Feature flags
+  - Account state (etc. logged in/out, sync engines enabled)
+  - Number of tabs open
+  - Size of Places database
+  - Active experiments
+  - Permissions are granted to the app
+  - WorkManager statue (e.g. scheduled/active/running)
+
+## Unresolved questions
+
+- Can we allow users to share logs with PII?
+  - This requires PII-aware logging and requires auditing our logs to ensure this.
```

Comment:
Regardless of this RFC, I think we should define some logging guidelines for us (and Fenix) in a separate RFC. Right now it's the wild west and there's no rule what is acceptable and at what level.

Reply:
🤠